internal/plan/plan.go

```diff
@@ -184,6 +187,76 @@ func (s *Service) Equal(other *Service) bool {
 	return reflect.DeepEqual(s, other)
 }
 
+// GetCommand overrides the default arguments (inside [ ]) if
+// CmdArgs is present and returns the command as a stream of strings.
+func (s *Service) GetCommand() ([]string, error) {
+	args, err := shlex.Split(s.Command)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse service %q command: %s", s.Name, err)
+	}
+	fargs := make([]string, 0)
+	for _, arg := range args {
+		if arg == "[" || arg == "]" {
+			if len(s.CmdArgs) > 0 {
+				break
+			} else {
+				continue
+			}
+		}
+		fargs = append(fargs, arg)
+	}
+	for _, arg := range s.CmdArgs {
+		fargs = append(fargs, arg)
+	}
+	return fargs, nil
+}
+
+// GetCommandStr takes in the output of GetCommand and
+// returns the command as a string in good print format
+func (s *Service) GetCommandStr(cmd []string) string {
+	f := func(str string) string {
+		for _, c := range str {
+			if c < '!' {
+				return strconv.Quote(str)
+			}
+		}
+		return str
+	}
+	args := make([]string, 0)
+	for _, arg := range cmd {
+		args = append(args, f(arg))
+	}
+	return strings.Join(args, " ")
+}
```

Comment:
This seems like a halfway house that's neither unquoted nor properly-quoted. Seeing this is just used for logging I guess we could just log the slice with `%q` and be done with it.

It's a pity there's no `shlex.Join` function! Make it wouldn't be hard to make one based on https://github.com/python/cpython/blob/fd155b91392c5bd238781bb11e040cf46c94725c/Lib/shlex.py#L323C31-L334 -- should only be a few lines of code in Go too.

Reply:
We could add this to our share x-go library?

Reply:
Yeah, something like `shlex.Join` would be nice indeed. I will try to write that up, and probably make a PR to x-go. Thanks.
I am ditching the `GetCommandStr` function, it was indeed a means for somewhat better logging.